src/sentry_core.c

```diff
@@ -75,10 +76,24 @@ sentry_init(sentry_options_t *options)
     SENTRY_DEBUGF("using database path \"%" SENTRY_PATH_PRI "\"",
         options->database_path->path);
 
+    // try to create and lock our run folder as early as possibly, since it is
+    // fallible. since it does locking, it will not interfere with run folder
+    // enumeration.
+    options->run = sentry__run_new(options->database_path);
+    if (!options->run) {
+        sentry_options_free(options);
+        return 1;
```

Comment:
Potentially there could be a way to use part of the SDK even without a run folder. It's fair to assume that we need it, though.

Reply:
hm… crashpad needs it for the msgpack files, breakpad needs it in case of a hard crash, as does the inproc backend. Not sure it makes sense to proceed without a run folder.